README.md

```diff
@@ -1,36 +1,134 @@
-# Decentralized ISO20022
-Financial Inclusion with decentralized ISO20022 as a microservice.
+# Decentralized ISO20022 microservice
+Financial Inclusion with decentralized ISO20022 as a bidirectional microservice. It consists of two components, the internal API and the external API.
+The internal API should be kept private and only accessible by your own financial institution, which could be done using for example a
+VPN or a firewall. Its purpose is to validate outgoing ISO 20022 messages before they are sent to a remote financial institution. The external API is
+publicly available for remote financial institutions and can accept and validate incoming ISO 20022 messages before forwarding them
+to an internal handler service.
+
+## Table of Contents
+ - [1. System Overview](#1-system-overview)
+ - [2. Sequence Diagram](#2-sequence-diagram)
+ - [3. Use In Settlement And Payments](#3-use-in-settlement-and-payments)
+ - [4. Technologies](#4-technologies)
+ - [5. Demo](#5-demo)
+ - [6. How To Use](#6-how-to-use)
+   - [6.1. Endpoints](#61-endpoints)
+   - [6.2. Certificates](#62-certificates)
+   - [6.3. Sending Messages](#63-sending-messsages)
+   - [6.4. Receiving Messages](#64-receiving-messages)
+   - [6.5. Configuration](#65-configuration)
+ - [7. Setup](#7-setup)
+   - [7.1. Using Linux](#71-using-linux)
+   - [7.2. Using Windows](#72-using-windows)
+   - [7.3. Using Docker](#73-using-docker)
+ - [8. Software Architecture](#8-software-architecture)
 
-## Technologies
-The technologies used in the project. 
-- [Spring Boot](https://github.com/spring-projects/spring-boot) for the framework.
-- [Prowide-ISO20022](https://github.com/prowide/prowide-iso20022) for parsing, creating and manipulating ISO20022 messages.
 
-## System overview
-An overview of the system.
+## 1. System overview
+An overview of the system, using VPNs to protect the internal APIs.
 ![Image of the system overview](images/system-overview.png)
 
-## Sending and recieving a pacs.008 message
-A sequence diagram for sending and recieving a pacs.008 message with the microservice. Communication service represents an instance of the microservice.
+## 2. Sequence diagram
+A sequence diagram for sending and receiving a pacs.008 message with the microservice. !!!!!!!!!!!!!!!!!!!!!!!Communication service represents an instance of the microservice.!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
Why are there so many exclamation marks? Is it intentional?

Reply:
It was to remind myself of something that is not relevant anymore, I removed them.